MVCDemos/Controllers/ApiControllers/DataGridRowReorderingController.cs

```diff
@@ -28,32 +28,26 @@ public class DataGridRowReorderingController : ApiController {
             var oldOrderIndex = task.OrderIndex;
 
             JsonConvert.PopulateObject(values, task);
+            var newOrderIndex = task.OrderIndex;
 
             Validate(task);
 
-            _context.Tasks.Remove(task);
-            var sortedTasks = _context.Tasks.OrderBy(t => t.OrderIndex).ToList();
+            if(oldOrderIndex != newOrderIndex) {
+                task.OrderIndex = oldOrderIndex;
+                var sortedTasks = _context.Tasks.OrderBy(t => t.OrderIndex).ToList();
```

Comment:
If we use ToList here why do we get an element using ElementAt instead of simple sortedTasks[]